### PR TITLE
Use SonarLint core 1.1

### PR DESCRIPTION
**Pull request by dbmeneses**
With this change, we can use IntelliJ testing framework again later

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/analysis/DefaultSonarLintFacade.java [hunk 1]
```diff
@@ -28,9 +28,8 @@
 import org.sonarlint.intellij.config.project.SonarLintProjectSettings;
 import org.sonarlint.intellij.ui.SonarLintConsole;
 import org.sonarlint.intellij.util.SonarLogOutput;
-import org.sonarsource.sonarlint.core.AnalysisConfiguration;
-import org.sonarsource.sonarlint.core.IssueListener;
-import org.sonarsource.sonarlint.core.SonarLintClient;
+import org.sonarsource.sonarlint.core.SonarLintClientImpl;
+import org.sonarsource.sonarlint.core.client.api.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/DefaultSonarLintFacade.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/intellij/issue/IssueMatcher.java [hunk 2]
```diff
@@ -32,7 +32,8 @@
 import com.intellij.psi.PsiFile;
 import com.intellij.psi.PsiManager;
 import com.intellij.psi.PsiWhiteSpace;
-import static org.sonarsource.sonarlint.core.IssueListener.Issue;
+import org.sonarsource.sonarlint.core.client.api.ClientInputFile;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/issue/IssueMatcher.java]: MINOR Remove this unused import 'org.sonarsource.sonarlint.core.client.api.ClientInputFile'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/issue/IssueMatcher.java [hunk 3]
```diff
@@ -32,7 +32,8 @@
...
+import org.sonarsource.sonarlint.core.client.api.Issue;
 
 import javax.annotation.Nullable;
 import java.nio.file.Path;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/issue/IssueMatcher.java]: MINOR Remove this unused import 'java.nio.file.Path'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/issue/IssueProcessor.java [hunk 4]
```diff
@@ -32,7 +32,8 @@
 import org.sonarlint.intellij.messages.AnalysisResultsListener;
 import org.sonarlint.intellij.ui.SonarLintConsole;
 import org.sonarlint.intellij.util.SonarLintUtils;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/issue/IssueProcessor.java]: MINOR Remove this unused import 'org.sonarlint.intellij.util.SonarLintUtils'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalysisConfigurator.java [hunk 5]
```diff
@@ -104,15 +102,15 @@ private static Charset getEncoding(Project p, @Nullable VirtualFile f) {
...
 
-  private static List<InputFile> getInputFiles(Project p, ModuleRootManager moduleRootManager, Collection<VirtualFile> filesToAnalyze) {
+  private static List<ClientInputFile> getInputFiles(Project p, ModuleRootManager moduleRootManager, Collection<VirtualFile> filesToAnalyze) {
     Collection<String> testFolderPrefix = findTestFolderPrefixes(moduleRootManager);
-    List<InputFile> inputFiles = new LinkedList<>();
+    List<ClientInputFile> inputFiles = new LinkedList<>();
 
     for (VirtualFile f : filesToAnalyze) {
       boolean test = isTestFile(testFolderPrefix, f);
       Path path = Paths.get(f.getPath());
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalysisConfigurator.java]: MAJOR Remove this useless assignment to local variable "path". [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1854)
MAJOR Remove this unused "path" local variable. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1481)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 3 major
* MINOR 4 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'com.intellij.openapi.vfs.VirtualFileSystem'.](https://github.com/SonarSource/sonarlint-intellij/blob/f8e429e18513ff637d9b64552fa0c18ee61ebffb/src/main/java/org/sonarlint/intellij/issue/IssueMatcher.java#L29) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalysisConfigurator.java [hunk 6]
```diff
@@ -42,10 +42,8 @@
...
 import org.sonarlint.intellij.ui.SonarLintConsole;
-
-import static org.sonarsource.sonarlint.core.AnalysisConfiguration.InputFile;
-
-import org.sonarsource.sonarlint.core.IssueListener;
+import org.sonarsource.sonarlint.core.client.api.IssueListener;
+import org.sonarsource.sonarlint.core.client.api.ClientInputFile;
 
 import java.nio.charset.Charset;
 import java.nio.file.Path;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalysisConfigurator.java]: MINOR Remove this unused import 'java.nio.file.Path'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'java.nio.file.Paths'.](https://github.com/SonarSource/sonarlint-intellij/blob/e14ec8a8ba9e7fca42464436597d2775c3a1664f/src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalysisConfigurator.java#L50) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM

**Pull request closed without merging**